### PR TITLE
rladmin reference (Doc-83, DOC-210, DOC-606)

### PR DESCRIPTION
**Pull request by kaitlynmichael**

### CODE REVIEW

**Comment by AlonMagrafta** [comment]
@kaitlynmichaelredislabs - While I review this, how are we planning on keeping this up-to-date?

**Comment by kaitlynmichaelredislabs** [comment]
@AlonMagrafta we want any public documentation to always be up-to-date. I already know this page is missing `rladmin cluster config saslauthd_ldap_conf`. I've created DOC-606 to track that.

**Comment by kaitlynmichaelredislabs** [comment]
Please try to find time to review this PR this week; I am aiming to merge this on Friday 3/12.

File: content/rs/references/rladmin.md [hunk 1]
```diff
@@ -1,30 +1,697 @@
...
+`rladmin failover` performs a failover for master shards of a specific database, and promotes their respective replicas to new masters. The `immediate` flag performs the failover without verifying the replica shards are in full sync with the master shards.
+
+```text
+rladmin failover 
+        [ db <db:id | name> ] 
+        shard <id1 .. idN> 
+        [ immediate ]
+```
+
+### `help`
```
Comment (amiramm) [inline content/rs/references/rladmin.md:233]: @kaitlynmichaelredislabs If I may suggest - can we start with the Help and Info commands and then move up in order of complexity/impact/danger?

File: content/rs/references/rladmin.md [hunk 2]
```diff
@@ -1,30 +1,697 @@
...
+rladmin verify balance [ node <id> ]
+```
+
+`rladmin verify rack_aware` prints a cluster rack aware verification report.
+
+```text
+rladmin verify rack_aware
+```
+
+## Use the rladmin shell
```
Comment (amiramm) [inline content/rs/references/rladmin.md:688]: I would lead with this info.

File: content/rs/references/rladmin.md [hunk 3]
```diff
@@ -1,30 +1,697 @@
 ---
```
Comment (amiramm) [inline content/rs/references/rladmin.md:1]: General comment - @AlonMagrafta @kaitlynmichaelredislabs thoughts about how we indicate which commands are `dangerous` - could lead to data loss or `caution` - can have other adverse impacts - if at all?
Some of these are required for setup and regular maintenance while some are advanced and shouldn't be used without consulting Redis Labs support.

File: content/rs/references/rladmin.md [hunk 4]
```diff
@@ -1,30 +1,697 @@
...
+```text
+rladmin info db <db:id | name>
+```
+
+```text
+rladmin info proxy <id | all>
+```
+
+```text
+rladmin info cluster <id>
```
Comment (amiramm) [inline content/rs/references/rladmin.md:250]:
```suggestion
rladmin info cluster
```
I don't think `rladmin info cluster` accepts additional parameters

**Review by amiramm — CHANGES_REQUESTED** [review]
Thanks for documenting this! I've made some comments, mostly around how we approach this type of document - you may have already discussed this offline.

File: content/rs/references/rladmin.md (hunk 1 above)
Comment (kaitlynmichaelredislabs) [reply content/rs/references/rladmin.md:233]: That's an interesting question. If they are organized by difficulty and not alphabetically, this page might be more difficult to navigate. Do we know if there is an industry standard for how to organize command reference material?

File: content/rs/references/rladmin.md (hunk 3 above)
Comment (kaitlynmichaelredislabs) [reply content/rs/references/rladmin.md:1]: I think we should mention any potential danger of data loss. I just need to know which commands are "dangerous" and how data loss or any other potential pitfalls can be avoided.

File: content/rs/references/rladmin.md (hunk 2 above)
Comment (kaitlynmichaelredislabs) [reply content/rs/references/rladmin.md:688]: I don't think the rladmin shell adds much value over just running individual rladmin commands besides the fact that it saves you a few keystrokes. Are there any capabilities the rladmin shell gives users that I missed in the list below?

Comment (amiramm) [reply content/rs/references/rladmin.md:688]: I think that it is much easier and user-friendly to run from the shell, especially for beginner users that may not even know that it is an option.

File: content/rs/references/rladmin.md (hunk 1 above)
Comment (amiramm) [reply content/rs/references/rladmin.md:233]: Here are a couple of references I'm familiar with and I think are pretty well organized.
https://docs.microsoft.com/en-us/cli/azure/get-started-with-azure-cli
https://docs.openshift.com/container-platform/4.7/cli_reference/openshift_cli/getting-started-cli.html

I really like how the Azure guide explains what it is, getting started, and then points to popular uses. Maybe we need to separate these from the alphabetized reference guide.

File: content/rs/references/rladmin.md (hunk 3 above)
Comment (AlonMagrafta) [reply content/rs/references/rladmin.md:1]: It's a good idea. We will need a thorough review of it to be able to mark them all

File: content/rs/references/rladmin.md [hunk 5]
```diff
@@ -1,30 +1,786 @@
...
 description:
 weight: $weight
 alwaysopen: false
 categories: ["RS"]
 aliases: /rs/references/cli-reference/rladmin/
 ---
-Redis Enterprise Software (RS) includes rladmin, a command-line interface (CLI)
-that you can use to do advanced administrative tasks.
-Some of these tasks you can also do through the management web UI.
+`rladmin` is a command-line utility for performing administrative tasks such as failover, migrate, and bind on a Redis Enterprise Software cluster. Some of these tasks can also be performed through the Management UI and some are unique to the `rladmin` CLI tool.
```
Comment (AlonMagrafta) [inline content/rs/references/rladmin.md:9]:
```suggestion
`rladmin` is a command-line utility for performing administrative tasks such as failover, migrate, and bind an endpoint on a Redis Enterprise Software cluster as well as cluster and database/s configuration updates. Some of these tasks can also be performed through the Management UI and some are unique to the `rladmin` CLI tool.
```

File: content/rs/references/rladmin.md [hunk 6]
```diff
@@ -1,30 +1,786 @@
...
+| Command (for configured clusters) | Description |
+| - | - |
+| `cluster certificate` | Sets the cluster certificate |
+| [`cluster config`](#cluster-config) | Updates configuration for the cluster |
+| [`cluster reset_password`](#cluster-reset_password) | Changes the password for a given email |
+| [`cluster stats_archiver`](#cluster-stats_archiver) | Enables/disables stats archiving |
+| [`cluster debug_info`](#cluster-debug_info) | Creates a support package |
+| `cluster running_actions` | Lists all active tasks |
+
+| Command (for non-configured clusters) | Description |
```
Comment (AlonMagrafta) [inline content/rs/references/rladmin.md:71]: I advise to move all these commands together with their extended description (currently lines 149-257) to a separate dedicated section about uninitialized clusters/nodes

File: content/rs/references/rladmin.md [hunk 7]
```diff
@@ -1,30 +1,786 @@
...
+| rack_id | Rack ID of the rack |
+| override_rack_id | Manually overrides the existing default rack ID |
+| flash_enabled | Enables flash storage |
+| flash_path | Path to flash storage location (defaults to /var/opt/redislabs/flash) |
+| addr | Internal IP addresses of the node |
+| external_addr | External IP addresses of the node |
+
+#### `cluster services`
+
+`rladmin cluster services` enables or disables selected cluster services.
```
Comment (AlonMagrafta) [inline content/rs/references/rladmin.md:261]: https://docs.redislabs.com/latest/rs/administering/troubleshooting/disabling-services/

File: content/rs/references/rladmin.md [hunk 8]
```diff
@@ -1,30 +1,786 @@
...
+
+```text
+rladmin cluster services [ cm_server | crdb_coordinator | crdb_worker | mdns_server | pdns_server | saslauthd | stats_archiver <enabled | disabled> ]
+```
+
+### `failover`
```
Comment (AlonMagrafta) [inline content/rs/references/rladmin.md:267]: @AlonMagrafta  : Mark to myself, I reviewed this PR to this point, not including the extended part of cluster create+recover+join

**Review by AlonMagrafta — CHANGES_REQUESTED** [review]
Review of lines 1-300

File: content/rs/references/rladmin.md (hunk 5 above)
Comment (kaitlynmichael) [reply content/rs/references/rladmin.md:9]:
```suggestion
`rladmin` is a command-line utility for performing administrative tasks such as failover, migration, and endpoint binding on a Redis Enterprise Software cluster. `rladmin` can also edit cluster and database configurations. Some of these tasks can also be performed through the Management UI and some are unique to the `rladmin` CLI tool.
```

File: content/rs/references/rladmin.md [hunk 9]
```diff
@@ -1,30 +1,786 @@
...
+| Command | Description |
+| - | - |
+| [rladmin](#use-the-rladmin-shell) | Enter the `rladmin` shell |
+| [bind](#bind) | Bind an endpoint |
+| [cluster](#cluster) | Cluster management commands |
+| [exit](#use-the-rladmin-shell)| Exit `rladmin` shell |
+| [failover](#failover) | Fail over master to replica |
+| [help](#help) | Show available commands or specific command usage |
+| [info](#info) | Show current parameters for a cluster, database, node, or proxy |
+| [migrate](#migrate) | Migrate elements between nodes |
```
[the remark this answers is not in the document]
Comment (kaitlynmichael) [reply content/rs/references/rladmin.md:22]: From the info I have, it looks like `rladmin migrate` only migrates shards and endpoints (not databases or nodes)

File: content/rs/references/rladmin.md [hunk 10]
```diff
@@ -1,30 +1,786 @@
...
+        certificate_file <certificate_file> 
+        [ key_file <certificate_file> ]
+```
+
+#### `cluster config`
+
+`rladmin cluster config` updates the cluster configuration.
+
+```text
+ rladmin cluster config
```
[the remark this answers is not in the document]
Comment (kaitlynmichael) [reply content/rs/references/rladmin.md:93]: @AlonMagrafta  Do you have the arguments and descriptions for `sentinel_cipher_suites`, `services`, `upgrade_mode`?

File: content/rs/references/rladmin.md [hunk 11]
```diff
@@ -1,30 +1,697 @@
...
+```
 
-    If you are a non-root user, you must add `/opt/redislabs/bin/` to your PATH environment variables
+| Proxy policy | Description |
+| - | - |
+| single | All traffic flows through a single proxy bound to the database endpoint (preferable in most cases)               |
+| all-master-shards | Multiple proxies, one on each master node (best for high traffic and multiple master shards)                     |
+| all-nodes | Multiple proxies, one on each node of the cluster (increases traffic in the cluster, only used in special cases) |
+| Legacy | Copies existing binding configuration from earlier versions
+
```
[the remark this answers is not in the document]
Comment (kaitlynmichael) [reply content/rs/references/rladmin.md:50]: I am going to push out the initial version of this page today and add links the next revision.

File: content/rs/references/rladmin.md (hunk 2 above)
Comment (kaitlynmichael) [reply content/rs/references/rladmin.md:777]: I will make sure to mention the shell when writing "how-to" articles. When I can expand each section into its own page, we will be able to add more detail.

File: content/rs/references/rladmin.md [hunk 12]
```diff
@@ -1,30 +1,786 @@
...
+## rladmin Commands
 
-- View the detailed status of the cluster including information about each node, each database, and each shard
-- Failover between a master and slave shard, or a master and slave endpoint
-- Migrate a specific shard or endpoint between nodes
-- Tune specific internal parameters of a database
-- Upgrade the version of a database to the most current version supported by the cluster
+| Command | Description |
+| - | - |
+| [rladmin](#use-the-rladmin-shell) | Enter the `rladmin` shell |
```
[the remark this answers is not in the document]
Comment (kaitlynmichael) [reply content/rs/references/rladmin.md:15]: These options are mentioned in the rladmin shell section. I want to keep this table as clean as possible.

File: content/rs/references/rladmin.md (hunk 7 above)
Comment (kaitlynmichael) [reply content/rs/references/rladmin.md:273]: I will add links during the DOC-616 version

File: content/rs/references/rladmin.md (hunk 6 above)
Comment (kaitlynmichael) [reply content/rs/references/rladmin.md:71]: I'll look into this change with DOC-617

File: content/rs/references/rladmin.md [hunk 13]
```diff
@@ -1,30 +1,786 @@
...
+
+| Command (for non-configured clusters) | Description |
+| - | - |
+| [`cluster create`](#cluster-create) | Creates a new cluster |
+| [`cluster join`](#cluster-join) | Adds a node to an existing cluster |
+| [`cluster recover_filename`](#cluster-recover_filename) | Recovers a cluster from a backup file |
+
+#### `cluster certificate`
+
+ `rladmin cluster certificate` sets the cluster certificate to a given file.
```
[the remark this answers is not in the document]
Comment (kaitlynmichael) [reply content/rs/references/rladmin.md:79]: This will be for DOC-616

File: content/rs/references/rladmin.md (hunk 3 above)
Comment (kaitlynmichael) [reply content/rs/references/rladmin.md:1]: I created DOC-618 for marking the "dangerous" commands

**Pull request merged**